### PR TITLE
Add linting workflow to CI

### PR DESCRIPTION
**Pull request by OliverStolzBO**
Run ESLint check when relevant files are changed.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# SPDX-FileCopyrightText: 2024 The PFDL VS Code Extension Contributors
```
Comment (marbre) [inline .github/workflows/lint.yml]:
```suggestion
# SPDX-FileCopyrightText: The PFDL VS Code Extension Contributors
```
No need to add a year.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# SPDX-License-Identifier: MIT
+
+name: Lint
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
```
Comment (marbre) [inline .github/workflows/lint.yml:10]:
```suggestion
  push:
    branches:
      - main
```
You don't want to run this on every push in every branch.

**Review by marbre — CHANGES_REQUESTED** [review]

**Comment by marbre** [comment]
How does this action work? Does it just run the linter (I am not familiar with typescript at all)? If yes, the linter will probably always pass if it isn't segfaulting. This means you need to check if there are changes to a prior / the checkin revision.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (OliverStolzBO) [reply .github/workflows/lint.yml:10]: As far as I understand @ptrdtznr, this is exactly the behaviour we want (linting on every push / branch)

**Comment by OliverStolzBO** [comment]
The command eslint ./ (triggered by npm run lint) should check the code for linting errors according to the rules defined in the [.eslintrc.js](https://github.com/iml130/pfdl-vscode-extension/blob/ci/linting-workflow/.eslintrc.js) and return an error code if errors are found. Therefore the job should fail when linting errors exist.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (marbre) [reply .github/workflows/lint.yml:10]: No, this is definitely not what you want ad we only have limited CI resources. Lint on push to main and on PRs should be sufficient. Do you really want to get an email on every commit your pushing to a feature / development branch?

**Comment by marbre** [comment]
> The command eslint ./ (triggered by npm run lint) should check the code for linting errors according to the rules defined in the [.eslintrc.js](https://github.com/iml130/pfdl-vscode-extension/blob/ci/linting-workflow/.eslintrc.js) and return an error code if errors are found. Therefore the job should fail when linting errors exist.

As you say *should*, you don't seem to be 100% confident. What about introducing a formatting error in an additional commit to test the pipeline? This commit can be dropped before merging.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (OliverStolzBO) [reply .github/workflows/lint.yml:10]: Okay I see.

**Comment by OliverStolzBO** [comment]
The workflow acts as expected (See [the workflow for the code containing a formatting error](https://github.com/iml130/pfdl-vscode-extension/actions/runs/7972780113/job/21765251139?pr=13))

**Review by marbre — APPROVED** [review]

**Pull request merged**